huaweicloud/services/iec/data_source_huaweicloud_iec_vpc_subnets.go

```diff
@@ -116,12 +119,15 @@ func dataSourceIECVpcSubnetIdsRead(d *schema.ResourceData, meta interface{}) err
 		allIDs[i] = item.ID
 	}
 	if err := d.Set("subnets", iecSubnets); err != nil {
-		return fmtp.Errorf("Error saving IEC subnets: %s", err)
+		return diag.Errorf("error saving IEC subnets: %s", err)
 	}
 
 	// set id
 	d.SetId(hashcode.Strings(allIDs))
-	d.Set("region", GetRegion(d, config))
 
-	return nil
+	mErr := multierror.Append(nil,
+		d.Set("region", cfg.GetRegion(d)),
```

Comment:
subnets should also wrapped in multierror

```suggestion
		d.Set("region", cfg.GetRegion(d)),
		d.Set("subnets", iecSubnets),
```